pkg/apiserver/rest/metrics.go

```diff
@@ -37,10 +38,17 @@ var DualWriterOutcome = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 	NativeHistogramBucketFactor: 1.1,
 }, []string{"mode", "name", "method"})
 
-func (m *dualWriterMetrics) init() {
+func (m *dualWriterMetrics) init(log logr.Logger) {
 	m.legacy = DualWriterLegacyDuration
 	m.storage = DualWriterStorageDuration
 	m.outcome = DualWriterOutcome
+	reg := prometheus.DefaultRegisterer
```

Comment:
With import of pkg/infra/metrics

```suggestion
	reg := metrics.ProvideRegisterer()
```

Reply:
there's a linting rule disallowing imports from grafana to the apiserver pkg, so I'd rather keep what we have here.

Reply:
Then take a prometheus.Registerer as a dependency/requirement for instantiating `dualWriterMetrics`?

Reply:
that was actually my first attempt (see [PR](https://github.com/grafana/grafana/pull/89003)), but I guess it's way more straightforward if I just use the default prometheus register, which it seems to work as a singleton for the whole application, than passing the register around for every entity